LibSerialize.lua

```diff
@@ -1478,113 +1460,65 @@ function LibSerialize:IsSerializableType(...)
-        end
-    end
-
-    return ser._flushWriter()
-end
-
 function LibSerialize:SerializeEx(opts, ...)
     local ser = CreateSerializer(opts)
     return serializeOperation(ser, ...)
 end
 
+function LibSerialize:Serialize(...)
+    return self:SerializeEx(nil, ...)
+end
+
 function LibSerialize:SerializeAsyncEx(opts, ...)
     local ser = CreateSerializer(opts, true)
 
-    if opts.yieldOnElapsedTime and not opts.timeFn then
-        error("Async Mode operation with yieldOnElapsedTime requires timeFn option")
-    end
-
     local thread = coroutine_create(serializeOperation)
-    local dots = {...}
+    local input = {...}
     -- return coroutine handler
     return function()
-        local co_success, result = coroutine_resume(thread, ser, unpack(dots))
+        local co_success, result = coroutine_resume(thread, ser, unpack(input))
         if not co_success then
-            error(result)
+            return true, false, result
         elseif coroutine_status(thread) ~= 'dead' then
-            return true
+            return false
         else
-            return false, result
+            return true, result
         end
     end
 end
 
 function LibSerialize:SerializeAsync(...)
-    return self:SerializeAsyncEx(defaultAsyncOptions, ...)
-end
-
-function LibSerialize:Serialize(...)
-    return self:SerializeEx(defaultSerializeOptions, ...)
-end
-
-local function deserializeOperation(deser)
-    -- Since there's only one compression version currently,
-    -- no extra work needs to be done to decode the data.
-    local version = deser:_ReadByte()
-    assert(version <= DESERIALIZATION_VERSION, "Unknown serialization version!")
-
-    -- Since the objects we read may be nil, we need to explicitly
-    -- track the number of results and assign by index so that we
-    -- can call unpack() successfully at the end.
-    local output = {}
-    local outputSize = 0
-
-    while deser._readerBytesLeft() > 0 do
-        outputSize = outputSize + 1
-        output[outputSize] = deser:_ReadObject()
-    end
-
-    if deser._readerBytesLeft() < 0 then
-        error("Reader went past end of input")
-    end
-
-    return unpack(output, 1, outputSize)
-end
-
-function LibSerialize:DeserializeValue(input, opts)
-    local deser = CreateDeserializer(input, opts or {})
-    return deserializeOperation(deser)
+    return self:SerializeAsyncEx(nil, ...)
 end
 
-function LibSerialize:DeserializeAsyncValue(input, opts)
-    if opts.yieldOnElapsedTime and not opts.timeFn then
-        error("Async Mode operation with yieldOnElapsedTime requires timeFn option")
-    end
+function LibSerialize:DeserializeValue(input, opts, async)
+    local deser = CreateDeserializer(input, opts, async)
 
-    local deser = CreateDeserializer(input, opts or {}, true)
+    if async then
+        function checkCoroutineResult(thread, co_success, ...)
+            if not co_success then
+                return true, false, select(1, ...)
```

Comment:
`select(1, ...)` is redundant since `select(N, ...)` returns all values from N (N, N+1, etc.). `return true, false, ...` is fine.